mumax3c/scripts/driver.py

```diff
@@ -48,8 +50,31 @@ def driver_script(driver, system, compute=None, **kwargs):
             mx3 += f"gammaLL = {gamma0/mm.consts.mu0}\n"
             mx3 += "doprecess = true\n"
 
+        zh_li_terms = system.dynamics.get(type=mm.ZhangLi)
+        if zh_li_terms:
+            u = (
+                zh_li_terms[0].u
+                if isinstance(zh_li_terms[0].u, df.Field)
+                else df.Field(
+                    mesh=system.m.mesh,
+                    dim=3,
+                    value=(1.0, 0.0, 0.0),
+                    norm=zh_li_terms[0].u,
+                )
+            )
+
+            j = -np.multiply(
+                u
+                * (mm.consts.e / (mm.consts.e * mm.consts.hbar / (2.0 * mm.consts.me))),
+                system.m.norm,
+            )
+            j.write("j.ovf")
+            mx3 += f"Xi = {system.dynamics.get(type=mm.ZhangLi)[0].beta}\n"
+            mx3 += "Pol = 1\n"  # Current polarization is 1.
```

Comment:
That is implicitly used in `u -> j` conversion, right?

Reply:
Typically, current density (`J`) and polarization (`Pol`) are independent variables. However, `u` is proportional to `J * Pol`. Since in ubermag we only define `u` one can not obtain separate values of `J` and `Pol`. To keep the calculation of `J` simple, I fixed the value of `Pol` to 1. (Mumax3 requires defining `J` and `Pol` separately)